### PR TITLE
Update dropdown_menu to fix on android switch system language TextEdi…

### PR DESCRIPTION
**Pull request by ganluo960214**
Update dropdown_menu to fix on android switch system language TextEditingController not change

### CODE REVIEW

File: packages/flutter/lib/src/material/dropdown_menu.dart [hunk 1]
```diff
@@ -319,12 +320,10 @@ class _DropdownMenuState<T> extends State<DropdownMenu<T>> {
     buttonItemKeys = List<GlobalKey>.generate(filteredEntries.length, (int index) => GlobalKey());
     _menuHasEnabledItem = filteredEntries.any((DropdownMenuEntry<T> entry) => entry.enabled);
 
-    final int index = filteredEntries.indexWhere((DropdownMenuEntry<T> entry) => entry.value == widget.initialSelection);
-    if (index != -1) {
-      _textEditingController.text = filteredEntries[index].label;
-      _textEditingController.selection =
-          TextSelection.collapsed(offset: _textEditingController.text.length);
-    }
+
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/dropdown_menu.dart:323]:
```suggestion
```

File: packages/flutter/lib/src/material/dropdown_menu.dart [hunk 2]
```diff
@@ -319,12 +320,10 @@ class _DropdownMenuState<T> extends State<DropdownMenu<T>> {
...
-    final int index = filteredEntries.indexWhere((DropdownMenuEntry<T> entry) => entry.value == widget.initialSelection);
-    if (index != -1) {
-      _textEditingController.text = filteredEntries[index].label;
-      _textEditingController.selection =
-          TextSelection.collapsed(offset: _textEditingController.text.length);
-    }
+    
+    _selectedValue = widget.initialSelection;
+    changeTextEditingControllerStateBySelectedValue();
+
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/dropdown_menu.dart:326]:
```suggestion
```

File: packages/flutter/lib/src/material/dropdown_menu.dart [hunk 3]
```diff
@@ -517,6 +512,15 @@ class _DropdownMenuState<T> extends State<DropdownMenu<T>> {
     }
     setState(() {});
   }
+
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/dropdown_menu.dart:515]:
```suggestion

```

**Comment by HansMuller** [comment]
This update, which will need its own regression test, is failing many existing tests. For example:

```
03:19 +4234 ~21 -1: /b/s/w/ir/x/w/flutter/packages/flutter/test/material/dropdown_menu_test.dart: The onSelected gets called only when a selection is made (variant: TargetPlatform.fuchsia)           
 ══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
 The following TestFailure was thrown running a test:
 Expected: 'Item 2'
   Actual: 'Item 0'
    Which: is different.
           Expected: Item 2
             Actual: Item 0
                          ^
            Differ at offset 5
 When the exception was thrown, this was the stack:
 #4      main.<anonymous closure> (file:///b/s/w/ir/x/w/flutter/packages/flutter/test/material/dropdown_menu_test.dart:1151:5)
 <asynchronous suspension>
 #5      testWidgets.<anonymous closure>.<anonymous closure> (package:flutter_test/src/widget_tester.dart:168:15)
 <asynchronous suspension>
 #6      TestWidgetsFlutterBinding._runTestBody (package:flutter_test/src/binding.dart:1013:5)
 <asynchronous suspension>
 <asynchronous suspension>
 (elided one frame from package:stack_trace)
 This was caught by the test expectation on the following line:
   file:///b/s/w/ir/x/w/flutter/packages/flutter/test/material/dropdown_menu_test.dart line 1151
 The test description was:
   The onSelected gets called only when a selection is made (variant: TargetPlatform.fuchsia)
 ════════════════════════════════════════════════════════════════════════════════════════════════════
 03:19 +4234 ~21 -2: /b/s/w/ir/x/w/flutter/packages/flutter/test/material/dropdown_menu_test.dart: The onSelected gets called only when a selection is made (variant: TargetPlatform.fuchsia) [E]       
   Test failed. See exception logs above.
   The test description was: The onSelected gets called only when a selection is made (variant: TargetPlatform.fuchsia)
   
 To run this test again: /b/s/w/ir/x/w/flutter/bin/cache/dart-sdk/bin/dart test /b/s/w/ir/x/w/flutter/packages/flutter/test/material/dropdown_menu_test.dart -p vm --plain-name 'The onSelected gets called only when a selection is made (variant: TargetPlatform.fuchsia)'

```

**Comment by QuncCccccc** [comment]
Thanks for the contribution. Just curious if there's any issue that this PR is trying to fix?

**Comment by ganluo960214** [comment]
maybe next week I try to fix it.

**Comment by ganluo960214** [comment]
@QuncCccccc it's a bit complicated. it's about 'flutter_localizations'. here is a demo `https://github.com/ganluo960214/flutter_dropdown_menu_demo`

1. when dropdown_menu selected a item. the sub widget text will show 'en'.
2. after change android system language
3. the dropdown_menu sub widget text would not change.

image

**Comment by ganluo960214** [comment]
for now. i try to change flutter source on pc. but so many errors. maybe runtime error. and how to run test.
image

**Comment by ganluo960214** [comment]
@QuncCccccc this test got problem. the DropdownMenuEntries are all TestMenu.mainMenu0. it that allow?

**Comment by ganluo960214** [comment]
@QuncCccccc line 1085 to 1088

**Comment by QuncCccccc** [comment]
> @QuncCccccc this test got problem. the DropdownMenuEntries's value are all TestMenu.mainMenu0. it that allow?

Sorry for the late responding! And thanks for catching this! This should be a typo. The values in menu should be unique.

**Comment by QuncCccccc** [comment]
Close this PR just because there's no update for a while. Please feel free to reopen it when you are ready! Thanks again for the contribution:)

**Pull request closed without merging**